File: code-signal-arcade-universe/python3/almost_increasing_list.py
def almost_increasing_list(l):
    """
    List -> Boolean, given a list, return true if removing one element results
    in a sorted list
    """
    for i in range(len(l)):
        c = l[:i] + l[i+1:]
        if c == sorted(c) and len(c) == len(set(c)):
            return True
    return False

File: code-signal-arcade-universe/clojure/almost_increasing_list.clj
(defn almost_increasing_list
  "List -> Boolean, given a list, return true if removing one element results
    in a sorted list"
  [[x y z & rst :as v]]
  (cond
    (nil? z) (apply <= (conj rst y x))
    (or (<= z y)
        (<= z x)) (or (apply < (conj rst z x))
                      (apply < (conj rst y x))
                      (apply < (conj rst z y)))
    :default (recur (rest v))))
